sql/core/src/test/scala/org/apache/spark/sql/CollationSQLExpressionsSuite.scala

```diff
@@ -2295,6 +2295,827 @@ class CollationSQLExpressionsSuite
+      val dataType = BooleanType
+      assert(sql(query).schema.head.dataType == dataType)
+    }
+  }
+
+  test("arrays_overlap supports collation") {
+    val collation = "UNICODE"
+    val query = s"SELECT arrays_overlap(array('a', 'b', 'c'), array('c', 'd', 'e'));"
+    withSQLConf(SqlApiConf.DEFAULT_COLLATION -> collation) {
+      checkAnswer(
+        sql(query),
+        Seq(
+          Row(true)
+        )
+      )
+      // check result row data type
+      val dataType = BooleanType
+      assert(sql(query).schema.head.dataType == dataType)
+    }
+  }
+
+  test("array_insert supports collation") {
+    val collation = "UNICODE"
+    val query = s"SELECT array_insert(array('a', 'b', 'c', 'd'), 5, 'e');"
+    withSQLConf(SqlApiConf.DEFAULT_COLLATION -> collation) {
+      checkAnswer(
+        sql(query),
+        Seq(
+          Row(Seq("a", "b", "c", "d", "e"))
+        )
+      )
+      // check result row data type
+      val dataType = ArrayType(StringType(collation), true)
+      assert(sql(query).schema.head.dataType == dataType)
+    }
+  }
+
+  test("array_intersect supports collation") {
+    val collation = "UNICODE"
+    val query = s"SELECT array_intersect(array('a', 'b', 'c'), array('b', 'c', 'd'));"
+    withSQLConf(SqlApiConf.DEFAULT_COLLATION -> collation) {
+      checkAnswer(
+        sql(query),
+        Seq(
+          Row(Seq("b", "c"))
+        )
+      )
+      // check result row data type
+      val dataType = ArrayType(StringType(collation), false)
+      assert(sql(query).schema.head.dataType == dataType)
+    }
+  }
+
+  test("array_join supports collation") {
+    val collation = "UNICODE"
+    val query = s"SELECT array_join(array('hello', 'world'), ' ');"
+    withSQLConf(SqlApiConf.DEFAULT_COLLATION -> collation) {
+      checkAnswer(
+        sql(query),
+        Seq(
+          Row("hello world")
+        )
+      )
+      // check result row data type
+      val dataType = StringType(collation)
+      assert(sql(query).schema.head.dataType == dataType)
+    }
+  }
+
+  test("array_position supports collation") {
+    val collation = "UNICODE"
+    val query = s"SELECT array_position(array('a', 'b', 'c', 'c'), 'c');"
+    withSQLConf(SqlApiConf.DEFAULT_COLLATION -> collation) {
+      checkAnswer(
+        sql(query),
+        Seq(
+          Row(3)
+        )
+      )
+      // check result row data type
+      val dataType = LongType
+      assert(sql(query).schema.head.dataType == dataType)
+    }
+  }
+
+  test("array_size supports collation") {
+    val collation = "UNICODE"
+    val query = s"SELECT array_size(array('a', 'b', 'c', 'c'));"
+    withSQLConf(SqlApiConf.DEFAULT_COLLATION -> collation) {
+      checkAnswer(
+        sql(query),
+        Seq(
+          Row(4)
+        )
+      )
+      // check result row data type
+      val dataType = IntegerType
+      assert(sql(query).schema.head.dataType == dataType)
+    }
+  }
+
+  test("array_sort supports collation") {
```

Comment:
NOTE(long output here, not related to collations):
```
scala> sql("select array_sort(array('a', 'B'))").show
warning: 1 deprecation (since 2.13.3); for details, enable `:setting -deprecation` or `:replay -deprecation`
+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
|array_sort(array(a, B), lambdafunction((IF(((namedlambdavariable() IS NULL) AND (namedlambdavariable() IS NULL)), 0, (IF((namedlambdavariable() IS NULL), 1, (IF((namedlambdavariable() IS NULL), -1, (IF((namedlambdavariable() < namedlambdavariable()), -1, (IF((namedlambdavariable() > namedlambdavariable()), 1, 0)))))))))), namedlambdavariable(), namedlambdavariable()))|
+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
|                                                                                                                                                                                                                                                                                                                                                                            [B, a]|
+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
```
Should we consider removing the default lambda function from the output?

Reply:
I suspect that this issue spans across the expressions spectrum in Spark?
interesting note either way - thanks! we can keep an eye on this and follow-up in the future